### PR TITLE
chore: replace deprecated Spring Security APIs usage

### LINKED ISSUE #16872: Replace usage of deprecated authorizeHttpRequests() in VaadinWebSecurity
### Describe your motivation

Spring Security 6.1 has deprecated the method with the message
```
Deprecated For removal in 7.0. Use [authorizeHttpRequests(Customizer)] instead
```

### PR DESCRIPTION
**Pull request by mcollovati**
Replaces deprecated Spring Security APIs marked for removal in 7.0

Fixes #16872

### CODE REVIEW

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinWebSecurity.java [hunk 1]
```diff
@@ -171,45 +165,47 @@ protected void configure(HttpSecurity http) throws Exception {
...
-        urlRegistry.requestMatchers(
-                getDefaultWebSecurityIgnoreMatcher(requestUtil.getUrlMapping()))
-                .permitAll();
-
-        // all other requests require authentication
-        urlRegistry.anyRequest().authenticated();
+        http.requestCache(cfg -> cfg.requestCache(vaadinDefaultRequestCache));
+
+        // AuthorizeHttpRequestsConfigurer<HttpSecurity>.AuthorizationManagerRequestMatcherRegistry
+        // urlRegistry =
```
Comment (czp13) [inline vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinWebSecurity.java:187]: Are these lines leftovers or helping the understand the changes? I am not sure, a bit seems out of context without seeing the old/changed code :). Maybe we could remove it.

Comment (mcollovati) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinWebSecurity.java:187]: Should be removed, thanks

Comment (czp13) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinWebSecurity.java]: Thank you!

**Review by czp13 — DISMISSED** [review]
I have checked all changes, and the main idea is to use lambdas instead of the old deprecated way. All configs and setups are there, checked. Plus hillaPath -> completeHillaPath variable rename happened.

All makes sense, good changes, thank you, Marco.

**Review by czp13 — APPROVED** [review]

**Pull request merged**